### PR TITLE
Recent sightings

### PR DESCRIPTION
**Pull request by littlemauli**
5 most recent sightings from the database are shown in a table.

### CODE REVIEW

**Review by Yvonne-Wong — COMMENTED** [review]
Looks good to me Mariam, I can't think of any suggestions 😊

File: ClientApp/src/Api/apiClient.ts [hunk 1]
```diff
@@ -55,6 +55,33 @@ export async function submitSighting(newSighting: NewSighting) {
...
+export interface RecentSightingResponseList{
+  recentSightingsList: RecentSightingResponse[]
+}
+
+export async function getRecentSightings(): Promise<RecentSightingResponseList> {
+  const response = await fetch(`/recentsightings`);
+  if (!response.ok) {
+    throw new Error(await response.json());
+  }
+
```
Comment (chloejonessoftwire) [inline ClientApp/src/Api/apiClient.ts]: got a new line here that you could delete

File: ClientApp/src/Pages/Home/Home.scss [hunk 2]
```diff
@@ -17,6 +17,20 @@
     height: 100%;
 }
 
+.sightings-table{
+    background-color: #265077;
```
Comment (chloejonessoftwire) [inline ClientApp/src/Pages/Home/Home.scss]: could you turn your colors to constants and add them to './Styles/constants';

File: ClientApp/src/Pages/Home/Home.test.tsx [hunk 3]
```diff
@@ -2,12 +2,95 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import { MemoryRouter } from 'react-router-dom';
 import { Home } from './Home';
+import { fireEvent, render, screen,act } from '@testing-library/react';
 
-it('renders without crashing', async () => {
+
+
```
Comment (chloejonessoftwire) [inline ClientApp/src/Pages/Home/Home.test.tsx]: lots of empty lines

File: Models/Response/RecentSightingsListResponse.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+         public List<SightingResponse> RecentSightingsList {get; set;}
+
+    }
+}
+
+
+
+        // public SightingResponse(Sighting sighting)
+        // {
+        //     _sighting = sighting;
```
Comment (chloejonessoftwire) [inline Models/Response/RecentSightingsListResponse.cs]: delete out commented things if they're not important

**Review by chloejonessoftwire — COMMENTED** [review]
few changes including using the constants emily and roshni defined! otherwise looks good

File: ClientApp/src/Api/apiClient.ts [hunk 5]
```diff
@@ -55,15 +55,39 @@ export async function submitSighting(newSighting: NewSighting) {
   }
 }
 
+export interface RecentSightingResponse {
```
Comment (ChristianMcc) [inline ClientApp/src/Api/apiClient.ts]: I think this interface is the same as `SightingResponse` which is the same as `Sighting` in this file - could we move the 2 extra interfaces and just use one please?

File: ClientApp/src/Pages/Home/GetSightings.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import React, { Component, useEffect, useState } from 'react';
+import './Home.scss';
+
+
+interface Sighting {
+  sighting: {
+    location: string,
```
Comment (ChristianMcc) [inline ClientApp/src/Pages/Home/GetSightings.tsx:7]: indentation here :)

File: ClientApp/src/Pages/Home/Home.tsx [hunk 7]
```diff
@@ -1,7 +1,19 @@
-import React, { Component } from 'react';
+import React, { Component, useEffect, useState } from 'react';
 import './Home.scss';
+import { GetSightings } from './GetSightings';
+import {getRecentSightings} from '../../Api/apiClient';
+import {RecentSightingResponse, RecentSightingResponseList} from '../../Api/apiClient';
 
 function Home() {
+
+  const [myData, setMyData] = useState<RecentSightingResponseList | null>(null);
```
Comment (ChristianMcc) [inline ClientApp/src/Pages/Home/Home.tsx]: could we call this recentSightings or something similar instead of myData please

**Review by ChristianMcc — COMMENTED** [review]
Looking really good, couple of small points then it's ready to merge!

File: ClientApp/src/Pages/Home/Home.test.tsx [hunk 8]
```diff
@@ -2,12 +2,91 @@ import React from 'react';
...
+        "quantity": "18",
+        "location": "Scotland",
+        "latitude": 123,
+        "longitude": -11,
+        "description": "spotted 20 orca",
+        "sightedAt": "2021-03-31T00:00:00",
+        "submittedByName": "Roshni",
+        "submittedByEmail": "r@r.com"
+      }
+
```
Comment (chloejonessoftwire) [inline ClientApp/src/Pages/Home/Home.test.tsx]: extra line here

File: ClientApp/src/Pages/Home/Home.test.tsx [hunk 9]
```diff
@@ -2,12 +2,91 @@ import React from 'react';
...
+it('renders homeform successfully', async () => {
   const div = document.createElement('div');
   ReactDOM.render(
     <MemoryRouter>
       <Home />
     </MemoryRouter>, div);
   await new Promise(resolve => setTimeout(resolve, 1000));
 });
+
+
```
Comment (chloejonessoftwire) [inline ClientApp/src/Pages/Home/Home.test.tsx]: few extra lines at the end

File: ClientApp/src/Pages/Home/Home.tsx (hunk 7 above)
Comment (ChristianMcc) [reply ClientApp/src/Pages/Home/Home.tsx]: I don't think this has been renamed yet :) have you pushed your changes?

**Review by ChristianMcc — APPROVED** [review]

**Pull request merged**